lib/tlscontext.c

```diff
@@ -321,30 +328,39 @@ tls_session_free(TLSSession *self)
   g_free(self);
 }
 
+EVTTAG *
+tls_context_format_location_tag(TLSContext *self)
+{
+  return evt_tag_str("location", self->location);
+}
+
 static gboolean
-_is_file_accessible(const gchar *fname)
+_is_file_accessible(TLSContext *self, const gchar *fname)
 {
   if (!fname)
     return FALSE;
   if (access(fname, R_OK) < 0)
     {
-      msg_error("Error opening TLS file",
+      msg_error("Error opening TLS related key or certificate file",
                 evt_tag_str("filename", fname),
-                evt_tag_errno("error", errno));
+                evt_tag_errno("error", errno),
```

Comment:
This is a general problem, but it is not safe to use errno this way. The malloc, strdup called from evt_tag_str can overwrite it. As the parameters are evaluated in any order, it is not even enough to move it to the first parameter.

As the original code had this defect too, and probably we will not run into this from practical point of view, I leave just a note here, and I create an issue to fix these in one swipe within the code. Maybe we can even assert if errno is passed directly to evt_tag_errno.